### PR TITLE
630 ac adapt richness

### PR DESCRIPTION
**Pull request by annacanal**
- Adapt richness measure for parametric format
- No need to change test_Measure
- All the test from workflows/Structural work now

Branch based on branch #628

### CODE REVIEW

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
1. Add comments in measure
2. Add methods to get info on name of parameter and its values
3. Correct errors in test functions (we need to discuss this before making changes)

**Comment by annacanal** [comment]
> 1. Add comments in measure
> 2. Add methods to get info on name of parameter and its values
> 3. Correct errors in test functions (we need to discuss this before making changes)

1 and 2 done. Will check the errors tomorrow morning

File: braph2/workflows/Structural/ComparisonST_WU.m [hunk 1]
```diff
@@ -212,7 +212,8 @@ function initialize_data(c, varargin)
                     varargin{:});
                 assert(iscell(c.getGroupValue(1)) && ...
                     isequal(size(c.getGroupValue(1)), [1, 1]) && ...
-                    all(cellfun(@(x) isequal(size(x), [1, 1]), c.getGroupValue(1))), ...
+                    all(cellfun(@(x) isequal(size(x, 1), 1), c.getGroupValue(1))) &&...
```
Comment (giovannivolpe) [inline braph2/workflows/Structural/ComparisonST_WU.m:215]: add space between && and ...
(also below)

Comment (annacanal) [reply braph2/workflows/Structural/ComparisonST_WU.m]: done everywhere

File: braph2/graph/measures/Richness.m [hunk 2]
```diff
@@ -204,6 +212,32 @@
...
+            % the values of the richness parameter.
+            
+            name = 'Richness threshold';
+            richness_threshold = get_from_varargin(1, 'RichnessThreshold', m.getSettings());
+            if richness_threshold > 0  % for positive threshold value, k = value
+                values = 1:1:richness_threshold;
+            else  % for negative threshold, k = max degree - threshold (default -1)
+                %k_level2 = max(deg) - abs(richness_threshold);  % problem knowing the degree
+                %values = 1:1:k_level2;
+                values = k_level; % will gives errors to negative thresholds now
```
[the remark this answers is not in the document]
Comment (annacanal) [reply braph2/graph/measures/Richness.m]: solved

File: braph2/workflows/Structural/MeasurementST_WU.m [hunk 3]
```diff
@@ -88,12 +100,17 @@ function initialize_data(m, varargin)
...
-            
+            % See also AnalysisST_WU.     
+
             atlases = m.getBrainAtlases();
             atlas = atlases{1};
 
             measure_code = m.getMeasureCode();
+            
+            m.parameter_values = get_from_varargin( ...
+                    {1}, ...  % 1 dimension minimum
```
Comment (giovannivolpe) [inline braph2/workflows/Structural/MeasurementST_WU.m]: []

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
1) correct sizes of defaults
2) check also 3rd dimension

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
Manage parameter values also in ComparisonST and RandomComparisonST as in MeasurementST

File: braph2/workflows/Structural/ComparisonST_WU.m [hunk 4]
```diff
@@ -175,6 +176,16 @@ function disp(c)
             
             confidence_interval_max = c.confidence_interval_max;
         end
+        function parameter_values = getParameterValues(c)
```
Comment (giovannivolpe) [inline braph2/workflows/Structural/ComparisonST_WU.m:179]: call it getMeasureParameterValues

File: braph2/workflows/Structural/RandomComparisonST_WU.m [hunk 5]
```diff
@@ -162,13 +163,23 @@ function disp(rc)
...
             %
-            % CONFIDENCE_INTERVAL_MAX = GETCONFIDENCEINTERVALMAX(C) 
+            % CONFIDENCE_INTERVAL_MAX = GETCONFIDENCEINTERVALMAX(RC) 
             % returns maximum value of the confidence interval
             % 
             % See also getP1, getP2, getConfidenceIntervalMin.
             
             confidence_interval_max = rc.confidence_interval_max;
         end
+        function parameter_values = getParameterValues(rc)
```
Comment (giovannivolpe) [inline braph2/workflows/Structural/RandomComparisonST_WU.m:173]: call it getMeasureParameterValues

File: braph2/workflows/Structural/AnalysisST_WU.m [hunk 6]
```diff
@@ -261,6 +265,8 @@
             
             difference = {value_group{1} - value_random{1}};
             
+            %TODO: change pvalue1 and pvalue2 functions or change how we
+            %send the variables
```
Comment (giovannivolpe) [inline braph2/workflows/Structural/AnalysisST_WU.m:269]: Is this done?

File: braph2/workflows/Structural/AnalysisST_WU.m [hunk 7]
```diff
@@ -289,6 +295,7 @@
                 'RandomComparisonST.p2', p2, ....
                 'RandomComparisonST.confidence_min', ci_lower, ...
                 'RandomComparisonST.confidence_max', ci_upper, ...
+                'RandomComparisonST.ParameterValuesLength', parameter_values_length, ...
```
Comment (giovannivolpe) [inline braph2/workflows/Structural/AnalysisST_WU.m:298]: should not exist

File: braph2/workflows/Structural/AnalysisST_WU.m [hunk 8]
```diff
@@ -390,6 +407,7 @@
                 'ComparisonST.p2', p2, ...
                 'ComparisonST.confidence_min', ci_lower, ...
                 'ComparisonST.confidence_max', ci_upper, ...
+                'ComparisonST.ParameterValuesLength', parameter_values_length1, ...
```
Comment (giovannivolpe) [inline braph2/workflows/Structural/AnalysisST_WU.m:410]: should not exist

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
small things

File: braph2/workflows/Structural/AnalysisST_WU.m (hunk 6 above)
Comment (annacanal) [reply braph2/workflows/Structural/AnalysisST_WU.m:269]: yes

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**